program/src/instructions/mplx_core/sell_nft_token_pool.rs

```diff
@@ -2,157 +2,73 @@
-        bump,
-        seeds::program = whitelist_program::ID
-    )]
-    pub whitelist: Box<Account<'info, WhitelistV2>>,
-
-    /// Optional account which must be passed in if the NFT must be verified against a
-    /// merkle proof condition in the whitelist.
-    /// CHECK: seeds and ownership are checked in assert_decode_mint_proof_v2.
-    pub mint_proof: Option<UncheckedAccount<'info>>,
-
-    /// The MPL core asset account.
-    /// CHECK: validated on instruction handler
-    #[account(mut)]
-    pub asset: UncheckedAccount<'info>,
-
-    /// CHECK: validated on instruction handler
-    pub collection: Option<UncheckedAccount<'info>>,
-
-    /// The shared escrow account for pools that have liquidity in a shared account.
-    /// CHECK: optional, manually handled in handler: 1)seeds, 2)program owner, 3)normal owner, 4)shared escrow acc stored on pool
-    #[account(mut)]
-    pub shared_escrow: Option<UncheckedAccount<'info>>,
-
-    /// The account that receives the maker broker fee.
-    /// CHECK: Must match the pool's maker_broker if passed in.
-    // validate() checks that the maker broker is passed in if the pool has a maker broker.
-    #[account(
-        mut,
-        constraint = pool.maker_broker != Pubkey::default() && maker_broker.key() == pool.maker_broker @ ErrorCode::WrongMakerBroker,
-    )]
-    pub maker_broker: Option<UncheckedAccount<'info>>,
-
-    /// The account that receives the taker broker fee.
-    /// CHECK: The caller decides who receives the fee, so no constraints are needed.
-    #[account(mut)]
-    pub taker_broker: Option<UncheckedAccount<'info>>,
-
-    /// The optional cosigner account that must be passed in if the pool has a cosigner.
-    // validate() checks that the cosigner is passed in if the pool has a cosigner.
-    #[account(
-        constraint = cosigner.key() == pool.cosigner @ ErrorCode::BadCosigner,
-    )]
-    pub cosigner: Option<Signer<'info>>,
-
-    /// The MPL Core program.
-    /// CHECK: address constraint is checked here
-    #[account(address = mpl_core::ID)]
-    pub mpl_core_program: UncheckedAccount<'info>,
-
-    /// The AMM program account, used for self-cpi logging.
-    pub amm_program: Program<'info, AmmProgram>,
-
-    /// The escrow program account for shared liquidity pools.
-    /// CHECK: address constraint is checked here
-    #[account(address = escrow::ID)]
-    pub escrow_program: Option<UncheckedAccount<'info>>,
+    pub core: MplCoreShared<'info>,
+
+    pub trade: TradeShared<'info>,
 
     /// The Solana system program.
     pub system_program: Program<'info, System>,
-    //
-    // ---- [0..n] remaining accounts for royalties transfer hook
+}
+
+impl<'info> Deref for SellNftTokenPoolCore<'info> {
+    type Target = MplCoreShared<'info>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.core
+    }
+}
+
+impl<'info> DerefMut for SellNftTokenPoolCore<'info> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.core
+    }
```

Comment:
Probably keep it consistent with other files where you don't use deref on just one shared but not the other.

Reply:
Ok, removed Deref impls